.github/workflows/guardiand-docker.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish guardiand development docker image
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: certusone/guardiand
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Should we use the default GITHUB_TOKEN permissions or generate a special key for this action and scope down it's permissions to a minimal set of scope?

Reply:
No need for that, I've specified in the workflow file that the token should only have write access to the packages (where it publishes): https://github.com/certusone/wormhole/pull/1332/files#diff-0646c51b93a786d348c46b78d911a1c9d066b8263e4f0dfc817839bdf86fd0c1R15-R17

reference: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
> You can use the `permissions` key in your workflow file to modify permissions for the GITHUB_TOKEN for an entire workflow or for individual jobs. This allows you to configure the minimum required permissions for a workflow or job. When the `permissions` key is used, all unspecified permissions are set to no access, with the exception of the metadata scope, which always gets read access.